tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+(APT). APT is a package *repository*; it contains both the **database** of the
+packages available in Ubuntu, and it contains the **packages** themselves.
+
+There are two commands we need for updating our system: `update` and `upgrade`,
+which we will always run in that order.
+
+### apt update
+
+The `apt update` command is about the **database**. Any bug fixes in a package
+(or new versions since your last update) will be stored in the metadata about
+that package in the database (the **package index**).
+
+When we run the `update` command it updates the APT database on our machine,
+fetching the newest available metadata from the package index:
+
+```bash
+sudo apt update
+```
+
+We will see an output like this:
+
+```text
+Hit:1 http://security.ubuntu.com/ubuntu noble-security InRelease
+Hit:2 http://archive.ubuntu.com/ubuntu noble InRelease
+Hit:3 http://archive.ubuntu.com/ubuntu noble-updates InRelease
+Hit:4 http://archive.ubuntu.com/ubuntu noble-backports InRelease
+Reading package lists... Done
+Building dependency tree... Done
+Reading state information... Done
+88 packages can be upgraded. Run 'apt list --upgradable' to see them.
+```
+
+As we can see, it checks ("hits") the various archives (**pockets**) that
+updates can come from for the 24.04 LTS release (`noble-security`, `noble`,
+`noble-updates` and `noble-backports` -- remember these, as we'll come back to
+them later). It has found some packages that can be upgraded to newer versions.
+If we want to see which packages those are, we can run the command hinted in
+the output:
+
+```bash
+apt list --upgradable
+```
+
+The output tells us:
+- the package name and where the update will come from
+  (e.g. `base-files/noble-updates`),
+- the most up-to-date package version available (e.g. `13ubuntu10.1`)
+- the hardware version the update is for (e.g. `amd64`), and
+- what package version is currently installed (e.g. `13ubuntu10`)
+
+The specific packages included in this list will change over time, but the
+output will be structured like this:
+
+```text
+Listing... Done
+base-files/noble-updates 13ubuntu10.1 amd64 [upgradable from: 13ubuntu10]
+bsdextrautils/noble-updates 2.39.3-9ubuntu6.1 amd64 [upgradable from: 2.39.3-9ubuntu6]
+bsdutils/noble-updates 1:2.39.3-9ubuntu6.1 amd64 [upgradable from: 1:2.39.3-9ubuntu6]
+cloud-init/noble-updates 24.2-0ubuntu1~24.04.2 all [upgradable from: 24.1.3-0ubuntu3.3]
+[...]
+```
+
+### apt upgrade
+
+The `apt upgrade` command is about the **packages** on your system. After we
+have updated the database by running `apt update` we can upgrade the packages
+to their newest versions by running:
+
+```bash
+sudo apt upgrade
+```
+
+The `upgrade` command looks at the metadata in the package index we just
+updated, finds the packages with available upgrades, and lists them for us. It
+also asks us to confirm if we want to do the upgrade. Let's type <kbd>Y</kbd>
+to confirm that we want to, and the upgrade will proceed. This may take a few
+minutes.
```

Comment:
You could also say they can press Enter for the default (Yes in this case because Y is capitalized)

I also think you should have this section start with "This command will ask you to confirm..." upfront. Not having that information be the first thing that came after the `upgrade` command is confusing. Or, you could inform the user before they run it that they'll be asked a prompt

Reply:
Reorganised the contents a bit to hopefully make the "you need to confirm" part a bit more visible, and added the Enter tip (I didn't know about that one!)